app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -505,15 +517,34 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 			}
 		} else {
 			loadPaginatedItems(directory.toListItem(), { !deferred.isCancelled }) {
-				deferred.complete(it)
+				val items = removeShufflePlayButtonMetadata(it)
+				deferred.complete(items)
 			}
 		}
 		return deferred.await()
 	}
 
-	override suspend fun search(query: String): List<MusicMetadata>? {
-		// requires a Web API call, not sure how to get the access token
-		return null
+	/**
+	 * Removes the shuffle play button [MusicMetadata] if it is present in the supplied list.
+	 *
+	 * When loading a list of tracks such as an album, a shuffle play button [MusicMetadata] object is
+	 * sometimes present. Call this method to get the list of [MusicMetadata]s that omits the shuffle
+	 * play button.
+	 */
+	private fun removeShufflePlayButtonMetadata(items: List<MusicMetadata>): List<MusicMetadata> {
```

Comment:
This tweak should be pulled out to a separate PR, because it doesn't really relate to the Search functionality.

Reply:
I added this refactoring since when an item such as an album is being browsed through the BrowseView the Shuffle Play button was present and needed to be removed. This problem was showing up when browsing a search result album type too